src/mesh/RTree.cpp

```diff
@@ -24,10 +26,25 @@ rtree::PtrRTree rtree::getVertexRTree(PtrMesh mesh)
   return tree;
 }
 
+PtrPrimitiveRTree rtree::getPrimitiveRTree(PtrMesh mesh)
+{
+  assertion(mesh, "Empty meshes are not allowed.");
+  auto iter = _primitive_trees.find(mesh->getID());
+  if (iter != _primitive_trees.end()) {
+    return iter->second;
+  } else {
+    auto treeptr = std::make_shared<PrimitiveRTree>(indexMesh(*mesh));
+    _primitive_trees.insert(std::make_pair(
+        mesh->getID(),
+        treeptr));
+    return treeptr;
+  }
+}
 
-void rtree::clear(Mesh & mesh)
+void rtree::clear(Mesh &mesh)
 {
   trees.erase(mesh.getID());
```

Comment:
`trees` should be renamed to `vertexTrees` or alike.

Reply:
I agree

Reply:
I'll rename `PtrRTree` to `PtrVertexRTree` in #156 as it would affect many places in the code.